src/framework/testing.ts

```diff
@@ -0,0 +1,66 @@
+import getPort from 'get-port'
+import { GraphQLClient } from 'graphql-request'
+import { App } from './app'
+import * as Layout from './layout'
+
+type AppClient = {
+  query: GraphQLClient['request']
+}
+
+export function createQueryClient(apiUrl: string): AppClient {
+  const client = new GraphQLClient(apiUrl)
+
+  return {
+    query(queryString, variables) {
+      return client.request(queryString, variables)
+    },
+  }
+}
+
+export type TestContext = {
+  app: AppClient
+  teardown: () => Promise<void>
+}
+
+/**
+ * Setup a test context providing utilities to query against your GraphQL API
+ */
+export async function setupTest(): Promise<TestContext> {
+  process.env.GRAPHQL_SANTA_STAGE = 'dev'
+
+  const port = await getPort({ port: getPort.makeRange(4000, 6000) })
+  const apiUrl = `http://localhost:${port}/graphql`
+
+  const app = require('./index').app as App
+
+  const oldServerStart = app.server.start
+
+  app.server.start = () => {
```

Comment:
https://github.com/prisma-labs/graphql-santa/issues/242#issuecomment-572189091

Reply:
`getPort` would still be async + that would only make it sync without prisma. With prisma we need to getDMMF, which is async

Reply:
Not to mention app module.

I still think we should keep it as building blocks though. Why start the server for them but then not stop it later? (the `teardown`).

So, async aside, I feel like the API, the level we're currently designing at, would be more consistent/predictable if it didn't try to abstract server start/stop.

Again, this boilerplate can be wrapped in a test helper function.

No?

Reply:
Completely agree, I'll make the changes!